internal/driver/controller.go

```diff
@@ -416,18 +427,30 @@ func (s controllerServerNoLocked) DeleteSnapshot(ctx context.Context, req *csi.D
 }
 
 // convertRequestCapacityBytes converts requestBytes and limitBytes to a valid capacity.
-func convertRequestCapacityBytes(requestBytes, limitBytes int64) (int64, error) {
+func convertRequestCapacityBytes(requestBytes, limitBytes int64, mount *csi.VolumeCapability_MountVolume) (int64, error) {
 	if requestBytes < 0 {
-		return 0, errors.New("required capacity must not be negative")
+		return 0, ErrNoNegativeRequestBytes
 	}
 	if limitBytes < 0 {
-		return 0, errors.New("capacity limit must not be negative")
+		return 0, ErrNoNegativeLimitBytes
 	}
 
 	if limitBytes != 0 && requestBytes > limitBytes {
-		return 0, fmt.Errorf(
-			"requested capacity exceeds limit capacity: request=%d limit=%d", requestBytes, limitBytes,
-		)
+		return 0, fmt.Errorf("%w: request=%d limit=%d", ErrRequestedExceedsLimit, requestBytes, limitBytes)
+	}
+
+	if mount != nil {
+		switch mount.GetFsType() {
+		case "xfs":
+			if limitBytes != 0 && limitBytes < topolvm.MinimumXfsSize {
+				return 0, ErrExceedsMinimumXfsSize
```

Comment:
So if I understand correctly 300Mi will just get rejected with an error? don't we want to round up instead and return success?

Reply:
No 300 Mi only gets rejected if you specified a limit that is 300Mi or below. If you just specify the request capacity as 200Mi then it would get increased under the hood to 300Mi. 

As per CSI Spec we must adhere to limits first, then requests. So if there is no limit we are free to expand but if the limit is lower than 300Mi this is now a new error case

Reply:
oh right, thanks for explaining, been a while since I've seen a PVC with limits on it